pkg/cli/upgrade.go

```diff
@@ -0,0 +1,104 @@
+package cli
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+
+	clusterv1beta1 "github.com/replicatedhq/embedded-cluster-kinds/apis/v1beta1"
+	"github.com/replicatedhq/embedded-cluster-operator/pkg/k8sutil"
+	"github.com/replicatedhq/embedded-cluster-operator/pkg/upgrade"
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+)
+
+// UpgradeCmd returns a cobra command for upgrading the embedded cluster operator.
+// It is called by KOTS admin console to upgrade the embedded cluster operator and installation.
+func UpgradeCmd() *cobra.Command {
+	var installationFile, localArtifactMirrorImage string
+
+	cmd := &cobra.Command{
+		Use:          "upgrade",
+		Short:        "Upgrade the embedded cluster operator",
+		SilenceUsage: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			fmt.Println("Upgrade command started")
+
+			cli, err := k8sutil.KubeClient()
+			if err != nil {
+				return fmt.Errorf("failed to create kubernetes client: %w", err)
+			}
+
+			installationData, err := readInstallationFile(installationFile)
+			if err != nil {
+				return fmt.Errorf("failed to read installation file: %w", err)
+			}
+
+			in, err := decodeInstallation(cmd.Context(), []byte(installationData))
+			if err != nil {
+				return fmt.Errorf("failed to decode installation: %w", err)
+			}
+
+			fmt.Printf("Upgrading to installation %s (k0s version %s)\n", in.Name, in.Spec.Config.Version)
+
+			err = upgrade.Upgrade(cmd.Context(), cli, in, localArtifactMirrorImage)
+			if err != nil {
+				return fmt.Errorf("failed to upgrade: %w", err)
+			}
+
+			fmt.Println("Upgrade command completed successfully")
+			return nil
+		},
+	}
+
+	cmd.Flags().StringVar(&localArtifactMirrorImage, "local-artifact-mirror-image", "", "Local artifact mirror image")
```

Comment:
this will most probably be included in the installation as well. but can updated later when that part is done.